Classes/EventListener/EventListenerLocator.php

```diff
@@ -68,50 +74,49 @@ public function getListeners(EventInterface $event): \Generator
      * @return array
      * @throws Exception
      * @Flow\CompileStatic
-     * @todo be sure that projector are called after all the other event listeners !!!
      */
     protected static function detectListeners(ObjectManagerInterface $objectManager): array
     {
         $listeners = [];
         /** @var ReflectionService $reflectionService */
         $reflectionService = $objectManager->get(ReflectionService::class);
+        /** @var EventTypeResolver $reflectionService */
+        $eventTypeService = $objectManager->get(EventTypeResolver::class);
         foreach ($reflectionService->getAllImplementationClassNamesForInterface(EventListenerInterface::class) as $listenerClassName) {
             foreach (get_class_methods($listenerClassName) as $methodName) {
+                $eventType = $eventClassName = null;
+
                 preg_match('/^when.*$/', $methodName, $matches);
                 if (!isset($matches[0])) {
                     continue;
                 }
-
                 $methodName = $matches[0];
                 $parameters = array_values($reflectionService->getMethodParameters($listenerClassName, $methodName));
-                $eventType = null;
 
                 switch (true) {
                     case count($parameters) === 1:
-                        $eventType = $parameters[0]['class'];
+                        $eventClassName = $parameters[0]['class'];
                         if (!$reflectionService->isClassImplementationOf($eventType, EventInterface::class)) {
```

Comment:
how is this supposed to work? `$eventType` is null at that point.. I'll push a follow-up